Cython/Utility/ObjectHandling.c

```diff
@@ -282,35 +282,46 @@ static CYTHON_INLINE PyObject *__Pyx_PyObject_GetItem(PyObject *obj, PyObject* k
 // //@requires: GetItemInt - added in IndexNode as it uses templating.
 
 #if CYTHON_USE_TYPE_SLOTS
-static PyObject *__Pyx_PyObject_GetIndex(PyObject *obj, PyObject* index) {
-    PyObject *runerr;
-    Py_ssize_t key_value;
-    PySequenceMethods *m = Py_TYPE(obj)->tp_as_sequence;
-    if (unlikely(!(m && m->sq_item))) {
-        PyErr_Format(PyExc_TypeError, "'%.200s' object is not subscriptable", Py_TYPE(obj)->tp_name);
-        return NULL;
-    }
-
-    key_value = __Pyx_PyIndex_AsSsize_t(index);
-    if (likely(key_value != -1 || !(runerr = PyErr_Occurred()))) {
-        return __Pyx_GetItemInt_Fast(obj, key_value, 0, 1, 1);
+static PyObject *__Pyx_PyObject_GetItem(PyObject *obj, PyObject* key) {
+    PyTypeObject* tp = Py_TYPE(obj);
+    PyMappingMethods *mm = tp->tp_as_mapping;
+    if (likely(mm && mm->mp_subscript)) {
+        return mm->mp_subscript(obj, key);
+    }
+
+    PySequenceMethods *sm = tp->tp_as_sequence;
+    if (likely(sm && sm->sq_item)) {
+        PyObject *runerr;
+        Py_ssize_t key_value;
+        key_value = __Pyx_PyIndex_AsSsize_t(key);
+        if (likely(key_value != -1 || !(runerr = PyErr_Occurred()))) {
+            return __Pyx_GetItemInt_Fast(obj, key_value, 0, 1, 1);
+        }
+        if (PyErr_GivenExceptionMatches(runerr, PyExc_OverflowError)) {
+            PyErr_Clear();
+            PyErr_Format(PyExc_IndexError, "cannot fit '%.200s' into an index-sized integer", Py_TYPE(key)->tp_name);
+        }
+        return NULL; 
+    }
+    
+#if PY_VERSION_HEX >= 0x03070000
+    if (likely(PyType_Check(obj))) {
+        PyObject *meth;
+        _Py_IDENTIFIER(__class_getitem__);
+        if (_PyObject_LookupAttrId(obj, &PyId___class_getitem__, &meth) < 0) {
+            return NULL;
+        }
+        if (meth) {
+            PyObject *result = __Pyx_PyObject_FastCall(meth, &key, 1);
```

Comment:
```suggestion
            PyObject *result = __Pyx_PyObject_CallOneArg(meth, key);
```
Does the same thing in the end (currently), but it's the explicit way to spell this in Cython's internal code.

Reply:
:+1: